### DISCUSSION TITLE
expire time settings

### DISCUSSION THREAD

**Discussion by DeepVasoya08** [devconcept/multer-gridfs-storage#524] [Q&A] [answered]
is there any way to set expire time for any file like images? here is code i'm using but it is not working

`
conn.once("open", () => {
  gfs = new mongoose.mongo.GridFSBucket(conn.db, {
    bucketName: "stories",
  });
});

const storage = new GridFsStorage({
  url: process.env.MONGO_URL,
  file: (req, file) => {
    return new Promise((resolve, reject) => {
      console.log(file);
      const filename = file.originalname;
      const fileInfo = {
        filename: filename,
        bucketName: "stories",
        createdAt: { type: Date, expires: 20, default: Date.now },
      };
      resolve(fileInfo);
      reject("server error!!");
    });
  },
  options:
});

const upload = multer({ storage });`

**Comment by devconcept** [comment, marked as answer]
Hello. Apologies for the very late response. The official support for chat was using gitter in the past but now that the repo is active again we will use github discussions and active monitor it. 

Custom fields should not be concern of your uploaded files. This should be stored in a database that then holds a reference to your file. This is true even if you are using the same mongodb database for data and files. Alternatively you can use metadata for this purpose.
